changelog.md

```diff
@@ -31,6 +31,25 @@
 
 - `options.UnpackError` is no longer a ref type and inherits from `System.Defect` instead of `System.ValueError`.
 
+- nre's `RegexMatch.{captureBounds,captures}[]`  no longer return `Option` or
+  `nil`/`""`, respectivly. Use the newly added `n in p.captures` method to
+  check if a group is captured, otherwise you'll recieve an exception.
```

Comment:
haven't thought about it much but is it possible to get a CT (instead of RT) error?

Reply:
Not unless the halting problem is solved. It's unknown whether a certain variable corresponds to a certain pattern. With constants it might be possible, but only when a group isn't optional (i.e. `(foo)?` will never fail at CT), and only if nim starts supporting compile-time FFI.

Reply:
/cc @flaviut 
that's not what I meant, but I worded it poorly; after this PR some of my tests broke, but they broke at RT instead of CT, which is not as good as if they failed at CT (because noone has 100% test coverage...)
how about the following:
* `m.captures[0]` returns the standard `Option[string]` instead of this non-standard `string` that throws on un-captured group?
that would've forced me to fix my code and guaranteed it would work 100% ; right now I have no such warrantee until my code hits such an issue in the wild

* furthermore, with `Option[string]`, I can also write (if my application doesn't care to distinguish bw nil vs empty string):
```
proc someOrEmpty(s: Option[string]): string = if s.some: s.get else: ""
m.captures[0].someOrEmpty
```